portfolio/src/main/webapp/script.js

```diff
@@ -48,14 +48,105 @@ function showMoreOrLessProjects() {
 }
 
 function getAndPrintComments() {
+
+    /*fetch('/data').then((response) => response.json()).then((comments) => {
     
-    fetch('/data').then((response) => response.json()).then((message) => {
-        var comments = "";
-        for(var i=0; i<message.length; i++){
-            comments+= "<br>"+message[i]+"<hr>";
-            console.log(message[i]);
-        } 
-        document.getElementById('comments-container').innerHTML = "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ comments;
-    });
-    
+    const commentListElement = document.getElementById('comments-container');
+    const buttonWithDropdown= "<div class='dropdown'>"+
+                    "<p><button class='commentSettings' onclick='commentSettings()'><i></i>&#9881;</button></p>"+
+                    "<div id='myDropdown' class='dropdown-content'>"+
+                        "<a href='#' onclick='deleteAllComments()'>Delete all comments</a>"+
+                        "<a href='#'>Set max no. of comments</a>"+
+                    "</div>"+
+                "</div>";
+    var commentsToBeAdded ="";
+    for(const comment of comments) {
+        commentsToBeAdded+= "<br>"+comment;
+    }
+    commentListElement.innerHTML= "<br><p style='font-size: larger;'><strong><em> Comments: </em></strong></p>"+ commentsToBeAdded+ buttonWithDropdown;
+  });*/
+  const ids = new Array();
+  const commentListElement = document.getElementById('comments-container');
+  fetch('/data').then(response => response.json()).then((comments) => {
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+      ids.push(comment.id);
+    })
+  });
+
+  const commentsDivElement = document.getElementById('comments-div');
+  //Creating a delete all button
+  const deleteAllButtonElement = document.createElement('button');
+  deleteAllButtonElement.innerText = 'Delete all comments';
+  deleteAllButtonElement.className = 'delete-all-button';
+  deleteAllButtonElement.addEventListener('click', () => {
```

Comment:
This feels wrong. If you have 100 comments you issue 100 individual requests. The server should provide a deleteAll function as this can be easier implemented on the server side.

I recommend adding a TODO and doing this in a next change.